cli/flox/src/commands/init/python.rs

```diff
@@ -539,23 +541,53 @@ impl Provider for PyProject {
 pub(super) struct Requirements {
     /// The latest version of python3 found in the catalogs
     python_version: String,
+    filename: String,
 }
 
 impl Requirements {
     fn detect(flox: &Flox, path: &Path) -> Result<Option<Self>> {
-        let requirements_txt = path.join("requirements.txt");
-
-        if !requirements_txt.exists() {
-            return Ok(None);
+        debug!("Detecting python requirements.txt at {:?}", path);
+        // NOTE: Does not match requirements files that have a prefix like `example_requirements.txt`
+        // See https://github.com/flox/flox/issues/1323
+        let pat = Regex::new(r#"^requirements\S*\.txt"#)?;
```

Comment:
blocking: Clippy is saying the `#` aren't needed
```suggestion
        let pat = Regex::new(r"^requirements\S*\.txt")?;
```